notificationapi_android_sdk/src/main/java/com/notificationapi/notificationapi_android_sdk/NotificationApi.kt

```diff
@@ -0,0 +1,86 @@
+package com.notificationapi.notificationapi_android_sdk
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.core.content.ContextCompat
+import com.notificationapi.notificationapi_android_sdk.models.NotificationApiConfig
+import com.notificationapi.notificationapi_android_sdk.models.NotificationApiCredentials
+import com.notificationapi.notificationapi_android_sdk.models.NotificationApiDeviceInfo
+import com.notificationapi.notificationapi_android_sdk.models.NotificationApiError
+
+class NotificationApi private constructor(private val context: Context) {
+    internal lateinit var deviceInfo: NotificationApiDeviceInfo
+
+    private lateinit var mConfig: NotificationApiConfig
+    internal val config: NotificationApiConfig
+        get() {
+            if (this::mConfig.isInitialized) {
+                return mConfig
+            }
+            throw NotificationApiError.missingCredentials
+        }
+
+    companion object {
+        internal const val TAG = "NotificationAPI"
+
+        internal const val NOTIFICATION_PERMISSION_REQUEST = 0
+
+        private const val CLIENT_ID = "notificationapi_client_id"
+        private const val USER_ID = "notificationapi_user_id"
+        private const val HASHED_ID = "notificationapi_hashed_user_id"
+
+        @SuppressLint("StaticFieldLeak")
+        private var mShared: NotificationApi? = null
+        val shared: NotificationApi
+            get() {
+                mShared?.let { return it }
+                throw NotificationApiError.uninitialized
+            }
+
+        fun initialize(context: Context) {
```

Comment:
This has to be called first. `NotificationApiActivity` and `NotificationApiService` will do this automatically